mobot/mobot_client/drop_session.py

```diff
@@ -0,0 +1,256 @@
+    SHIPPING_INFO_CONFIRMATION = 5
+    ALLOW_CONTACT_REQUESTED = 6
+    COMPLETED = 7
+
+    @classmethod
+    def active_states(cls):
+        return {
+                cls.NEW.value,
+                cls.WAITING_FOR_PAYMENT.value,
+                cls.WAITING_FOR_SIZE.value,
+                cls.WAITING_FOR_NAME.value,
+                cls.WAITING_FOR_ADDRESS.value,
+                cls.SHIPPING_INFO_CONFIRMATION.value,
+                cls.ALLOW_CONTACT_REQUESTED.value
+            }
+
+    @classmethod
+    def refundable_states(cls):
+        return {
+            cls.IDLE_AND_REFUNDABLE.value,
+            cls.WAITING_FOR_SIZE.value,
+            cls.WAITING_FOR_ADDRESS.value,
+            cls.WAITING_FOR_ADDRESS.value,
+            cls.WAITING_FOR_NAME.value,
+            cls.SHIPPING_INFO_CONFIRMATION.value,
+        }
+
+
+class DropType(enum.Enum):
+    AIRDROP = 0
+    ITEM = 1
+
+
+class BaseDropSession:
+    def __init__(self, store, payments, messenger):
+        self.store = store
+        self.payments = payments
+        self.messenger = messenger
+
+    @staticmethod
+    def get_advertising_drop():
+        drops_to_advertise = Drop.objects.filter(
+            advertisment_start_time__lte=timezone.now()
+        ).filter(start_time__gt=timezone.now())
+
+        if len(drops_to_advertise) > 0:
+            return drops_to_advertise[0]
+        return None
+
+    @staticmethod
+    def under_drop_quota(drop):
+        number_initial_drops_finished = DropSession.objects.filter(
+            drop=drop, state__gt=SessionState.READY_TO_RECEIVE_INITIAL.value
+        ).count()
+        return number_initial_drops_finished < drop.initial_coin_limit
+
+    def minimum_coin_available(self, drop):
+        unspent_pmob = self.payments.get_unspent_pmob()
+        return unspent_pmob >= (
+            drop.initial_coin_amount_pmob + int(self.payments.get_minimum_fee_pmob())
+        )
+
+    @staticmethod
+    def get_active_drop():
+        active_drops = Drop.objects.filter(start_time__lte=timezone.now()).filter(
+            end_time__gte=timezone.now()
+        )
+        if len(active_drops) == 0:
+            return None
+        return active_drops[0]
+
+    @staticmethod
+    def get_customer_store_preferences(customer, store_to_check):
+        try:
+            customer_store_preferences = CustomerStorePreferences.objects.get(
+                customer=customer, store=store_to_check
+            )
+            return customer_store_preferences
+        except (Exception,):
+            return None
+
+    def customer_has_store_preferences(self, customer):
+        try:
+            _ = CustomerStorePreferences.objects.get(
+                customer=customer, store=self.store
+            )
+            return True
+        except (Exception,):
+            return False
+
+    @staticmethod
+    def customer_has_completed_airdrop(customer, drop):
+        try:
+            _completed_drop_session = DropSession.objects.get(
```

Comment:
Same thing here - can we do a `models.OneToOneField` for the customer on DropSession?